relations/src/r1cs/constraint_system.rs

```diff
@@ -20,7 +20,7 @@ use ark_std::{
 // TODO: Think: should we replace this with just a closure?
 pub trait ConstraintSynthesizer<F: Field> {
```

Comment:
Probably we should rename this trait and add new methods like `generate_instance_assignment` and `generate_witness_assignment`